src/sentry/tasks/servicehooks.py

```diff
@@ -1,13 +1,98 @@
 from __future__ import absolute_import, print_function
 
 import six
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
 from time import time
 
+from sentry.api.serializers import serialize, app_platform_event
 from sentry.http import safe_urlopen
+from sentry.models import Group, SentryAppInstallation, ServiceHook
 from sentry.tasks.base import instrumented_task
 from sentry.utils import json
 from sentry.utils.http import absolute_uri
 
+# This is an extra, explicit, measure to ensure we only send events for
+# resource changes we deem necessary.
+ALLOWED_ACTIONS = (
+    'issue.created',
+)
+
+# We call some models by a different name, publically, than their class name.
+# For example the model Group is called "Issue" in the UI. We want the Service
+# Hook events to match what we externally call these primitives.
+RESOURCE_RENAMES = {
+    'group': 'issue',
+}
+
+
+@receiver(post_save, sender=Group)
+def resource_changed(sender, instance, created, **kwargs):
+    process_resource_change.delay(sender, instance.id, created)
+
+
+@instrumented_task('sentry.tasks.process_resource_change')
+def process_resource_change(sender, instance_id, created):
+    model = sender.__name__.lower()
+    model = RESOURCE_RENAMES.get(model, model)
+
+    instance = sender.objects.get(id=instance_id)
+
+    event = 'created' if created else 'updated'
+    action = u'{}.{}'.format(model, event)
+
+    if action not in ALLOWED_ACTIONS:
+        return
+
+    project = None
+
+    if isinstance(instance, Group):
+        project = instance.project
+
+    if not project:
+        return
+
+    servicehooks = ServiceHook.objects.filter(
+        project_id=project.id,
+    )
+
+    for servicehook in filter(lambda s: action in s.events, servicehooks):
+        # For now, these ``post_save`` callbacks are only valid for service
+        # hooks created by a Sentry App.
+        if not servicehook.created_by_sentry_app:
+            continue
+
+        payload = app_platform_event(
+            action,
+            SentryAppInstallation.objects.get(id=servicehook.actor_id),
+            serialize(instance),
+        )
+
+        send_request(servicehook, payload)
+
+
+def send_request(servicehook, payload):
+    from sentry import tsdb
+    tsdb.incr(tsdb.models.servicehook_fired, servicehook.id)
+
+    body = json.dumps(payload)
+
+    headers = {
+        'Content-Type': 'application/json',
+        'X-ServiceHook-Timestamp': six.text_type(int(time())),
+        'X-ServiceHook-GUID': servicehook.guid,
+        'X-ServiceHook-Signature': servicehook.build_signature(body),
+    }
+
+    safe_urlopen(
+        url=servicehook.url,
+        data=body,
+        headers=headers,
+        timeout=5,
+        verify_ssl=False,
```

Comment:
Why? Should sentryapps define whether they want SSL verification on their hooks?

Reply:
Yea, this was just to be backwards compatible with the current Service Hooks implementation. Gonna change this to be `True` for App Platform requests and `False` for old hooks. I don't understand why it's like this either, tbh.

Reply:
Seems like a WIP thing that grew roots.